spec/unit/facter/java_version_spec.rb

```diff
@@ -3,6 +3,7 @@
 describe Facter::Util::Fact do
   before {
     Facter.clear
+    Facter.fact(:osfamily).stubs(:value).returns("OpenBSD")
```

Comment:
the line could also be:
Facter.fact(:osfamily).stubs(:value).returns("DoesntMatter")

It's just to prevent the warning that I mentioned initially, and then failing the two tests.

Without that line, its trying to call `uname -s` which it doesn't like. It would fail the test like:

Facter::Util::Fact

java_version

returns java version when java present

example at ./spec/unit/facter/java_version_spec.rb:10 (FAILED - 1)

returns nil when java not present

should be nil (FAILED - 2)

Failures:

1) Facter::Util::Fact java_version returns java version when java present

Failure/Error: Facter.fact(:java_version).value.should == "1.7.0_71"

Mocha::ExpectationError:

unexpected invocation: Facter::Util::Resolution.exec('uname -s')

unsatisfied expectations:

- expected exactly once, not yet invoked: Facter::Util::Resolution.exec('java -version 2>&1')

- expected exactly once, not yet invoked: Facter::Util::Resolution.which('java')

I for myself find that hackish too, but I don't know a better way actually.
If you could point to something better, I'd be glad to change it.